tests/object_sync_test.go

```diff
@@ -1246,34 +1247,42 @@ func (c *CustomVarTestData) verify(t require.TestingT, logger *zap.Logger, db *s
 	require.NoError(t, err, "querying customvars")
 	defer func() { _ = rows.Close() }()
 
-	expectedSrc := c.Vars
-	if flat {
-		expectedSrc = c.VarsFlat
-	}
-
 	// copy map to remove items while reading from the database
-	expected := make(map[string]string)
-	for k, v := range expectedSrc {
-		expected[k] = v
+	expected := make(map[string]sql.NullString)
+	if flat {
+		for k, v := range c.VarsFlat {
+			expected[k] = v
+		}
+	} else {
+		for k, v := range c.Vars {
+			expected[k] = toDBString(v)
+		}
```

Comment:
If `v == ""` would `NULL` really be expected instead of `''`?

Reply:
`customvar.value` isn't nullable.

Reply:
So if there was a test with an empty string variable, with the current state of `toDBString()`, this would expect `NULL` in a column that can't be null. With the change I suggested in https://github.com/Icinga/icingadb/pull/601#discussion_r1234088452, it would be fine.